### ISSUE TITLE
Remove twisted.application.internet.UDPClient.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7702]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#7702|
|Type|enhancement|
|Created|2014-11-01 21:24:05Z|
|Branch|https://github.com/twisted/twisted/tree/remove-udpclient-7702|

It was deprecated in [#6468](https://github.com/twisted/twisted/issues/6468) (13.1), since it was broken. The deprecation period has ended, and the code doesn't work.

Attachments:

* [removal_7702.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/555/55508920d2dacc86c03dff2daa0a70e7a1a222f7/e044263774e3b19a3aafacc18c208c2aced5a9d2.patch) (1987 bytes) - added by eeshangarg on 2014-11-28 15:15:06Z - 
* [removal_7702_2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/555/55508920d2dacc86c03dff2daa0a70e7a1a222f7/a777b5881111e5157150b1f4f58f86302940baf2.patch) (3420 bytes) - added by eeshangarg on 2014-11-28 16:57:26Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7702 7702
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_remove_udpclient_7702 branches/remove-udpclient-7702
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1414877045578930 1414877045578930
changetime__1422085061548956 1422085061548956
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_1"></a>|@eeshangarg commented|
|-|-|

Hello!

I just added a patch for this(removal_7702.patch​). This patch removes all references to twisted.application.internet.UDPClient from the codebase.

If something needs to be improved or if I did something wrong, please let me know. I would love to work on it. :-)

Thanks & regards,
Eeshan Garg

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @eeshangarg|
|-|-|

Hi,

Thanks for the review.

All this post-mortem deprecation was added in [#6469](https://github.com/twisted/twisted/issues/6469).

Please check the merge commit and remove all lines introduced there
 
https://github.com/twisted/twisted/commit/70cc31a4b19e593c63e6e5da3d57ecae5f8ef5cf#diff-e1da3f98608a56680f93e0d2f0aa8b0dR23


You should also remove the import lines
```
+from twisted.python.deprecate import deprecatedModuleAttribute
+from twisted.python.versions import Version
```


-------
Then news line is a bit missleading. This just removed the deprecation warning, introduced in 13.1 ... but the real code was deprecated/removed in version 10

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince commented|
|-|-|

After applying this patch, `UDPClient` is still defined. (Note that that string doesn't appear in the source at the point of definition).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

As discussed over IRC, the UDPClient class is created at runtime by combining 'UDP' string with 'Client' string. I think that the patch for this branch should prevent this combination and raise ImportError

In case someone else is confuses (like I was) here is the code: 
https://github.com/twisted/twisted/blob/trunk/twisted/application/internet.py#L207-L209

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_5"></a>|@eeshangarg removed owner|
|-|-|

Hi!

I just submitted a new patch(removal_7702_2.patch) for this.

There were two tests that failed because of removing UDPClient. I modified the tests to exclude UDPClient.

I ran the tests before and after applying this patch. All of the tests passed.

You can no longer import twisted.application.internet.UDPClient now. :)

Thanks tom.prince, adiroiban for your guidance! If there is something I got wrong, please re-assign this ticket to me, I would love to work on this and close this ticket. :-)

Regards,
Eeshan Garg​

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

(In [43726]) Branching to remove-udpclient-7702.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph set owner to @glyph|
|-|-|

Thanks for submitting this patch, eeshangarg! Looks like everything is in order.  Buildbot is happy.  I'll land this now.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set status to `closed`|
|-|-|

(In [43728]) Merge remove-udpclient-7702: remove UDPClient

Author: eeshangarg

Reviewer: glyph

Fixes: [#7702](https://github.com/twisted/twisted/issues/7702)

Remove twisted.application.internet.UDPClient, deprecated since Twisted 13.1.0.
